R/FilteredDataset.R

```diff
@@ -30,7 +30,7 @@ FilteredDataset <- R6::R6Class( # nolint
     #' @return Object of class `FilteredDataset`, invisibly.
     #'
     initialize = function(dataset, dataname, keys = character(0), label = attr(dataset, "label", exact = TRUE)) {
-      check_simple_name(dataname)
+      checkmate::assert_character(dataname, min.len = 1, any.missing = FALSE)
```

Comment:
how dataname can be of length > 1? 
```suggestion
      checkmate::assert_string(dataname)
```

Reply:
I kept the previous assert in `check_simple_name`. I don't think it can in this initialize